lib/cc/services/github_pull_requests.rb

```diff
@@ -93,37 +102,44 @@ def add_comment
         body: COMMENT_BODY % @payload["compare_url"]
       }.to_json
 
-      http_post(comments_url, body)
+      post(comments_url, body) do |response|
+        doc = JSON.parse(response.body)
+        { id: doc["id"] }
+      end
     end
   end
 
   def receive_test_status
-    http_post(base_status_url("0" * 40), "{}")
-
-  rescue HTTPError => ex
-    if ex.status == 422 # response message: "No commit found for SHA"
-      { ok: true, message: "OAuth token is valid" }
-    else ex.status == 401 # response message: "Bad credentials"
```

Comment:
The 401s will get caught by the `WithErrorHandling` middleware and doesn't need to be a special case here anymore.